File: scripts/algorithms/F/Find Greatest Common Divisor of Array/Find Greatest Common Divisor of Array.java
// Runtime: 3 ms (Top 25.8%) | Memory: 42.56 MB (Top 95.7%)

class Solution {
    public int findGCD(int[] nums) {
        Arrays.sort(nums);
        int n=nums[nums.length-1];
        int result=nums[0];
        while(result>0){
            if(nums[0]%result==0 && n%result==0){
                break;
            }
            result--;
        }
        return result;
    }
}

File: scripts/algorithms/F/Find Greatest Common Divisor of Array/Find Greatest Common Divisor of Array.cpp
// Runtime: 20 ms (Top 7.45%) | Memory: 12.5 MB (Top 65.76%)
class Solution {
public:
    int findGCD(vector<int>& nums) {
        int mn = nums[0], mx = nums[0];
        for(auto n: nums)
        {
            // finding maximum, minimum values of the array.
            if(n > mx) mx = n;
            if(n < mn) mn = n;
        }

        for(int i = mn; i >= 1; i--)
        {
            // finding greatest common divisor (GCD) of max, min.
            if((mn % i == 0) && (mx % i == 0)) return i;
        }
        return 1;
    }
};